src/main/java/appeng/items/storage/ItemReadOnlyStorageCell.java

```diff
@@ -0,0 +1,240 @@
+				}
+			}
+		}
+	}
+
+	@Override
+	public int getBytes( final ItemStack cellItem )
+	{
+		return this.totalBytes;
+	}
+
+	@Override
+	public int getBytesPerType( final ItemStack cellItem )
+	{
+		return this.perType;
+	}
+
+	@Override
+	public int getTotalTypes( final ItemStack cellItem )
+	{
+		return 63;
+	}
+
+	@Override
+	public boolean isBlackListed( final ItemStack cellItem, final IAEItemStack requestedAddition )
+	{
+		return getReadOnly( cellItem );
+	}
+
+	@Override
+	public boolean storableInStorageCell()
+	{
+		return false;
+	}
+
+	@Override
+	public boolean isStorageCell( final ItemStack i )
+	{
+		return true;
+	}
+
+	@Override
+	public double getIdleDrain()
+	{
+		return this.idleDrain;
+	}
+
+	@Override
+	public String getUnlocalizedGroupName( final Set<ItemStack> others, final ItemStack is )
+	{
+		return GuiText.StorageCells.getUnlocalized();
+	}
+
+	@Override
+	public boolean isEditable( final ItemStack is )
+	{
+		return false;
+	}
+
+	@Override
+	public IInventory getUpgradesInventory( final ItemStack is )
+	{
+		return new CellUpgrades( is, 0 );
+	}
+
+	@Override
+	public IInventory getConfigInventory( final ItemStack is )
+	{
+		return new CellConfig( is );
+	}
+
+	@Override
+	public FuzzyMode getFuzzyMode( final ItemStack is )
+	{
+		return FuzzyMode.IGNORE_ALL;
+	}
+
+	@Override
+	public void setFuzzyMode( final ItemStack is, final FuzzyMode fzMode )
+	{
+
+	}
+
+	@Override
+	public ActionResult<ItemStack> onItemRightClick( final ItemStack stack, final World world, final EntityPlayer player, final EnumHand hand )
+	{
+		this.lockDrive( stack, player );
+		return new ActionResult<ItemStack>( EnumActionResult.SUCCESS, stack );
+	}
+
+	private boolean lockDrive( final ItemStack stack, final EntityPlayer player )
+	{
+		if( player.isSneaking() )
+		{
+			if( Platform.isClient() )
+			{
+				return false;
+			}
+
+			setReadOnly( stack, true );
+			return true;
+		}
+		return false;
+	}
+
+	@Override
+	public EnumActionResult onItemUseFirst( final ItemStack stack, final EntityPlayer player, final World world, final BlockPos pos, final EnumFacing side, final float hitX, final float hitY, final float hitZ, final EnumHand hand )
+	{
+		return this.lockDrive( stack, player ) ? EnumActionResult.SUCCESS : EnumActionResult.PASS;
+	}
+
+	@Override
+	public ItemStack getContainerItem( final ItemStack itemStack )
+	{
+		return AEApi.instance().definitions().materials().emptyStorageCell().maybeStack( 1 )
+				.orElseThrow( () -> new MissingDefinition( "Tried to use empty storage cells while basic storage cells are defined." ) );
+	}
+
+	@Override
+	public boolean hasContainerItem( final ItemStack stack )
+	{
+		return AEConfig.instance().isFeatureEnabled( AEFeature.ENABLE_DISASSEMBLY_CRAFTING );
+	}
+
+	@Override
+	public void setReadOnly(ItemStack is, boolean readOnly) {
+		final NBTTagCompound c = Platform.openNbtData( is );
+		c.setBoolean( "ReadOnly", readOnly );
+	}
+
+	@Override
+	public boolean getReadOnly(ItemStack is) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) '(' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) ')' is not preceded with whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)